### PR TITLE
Parking card

### PR DESCRIPTION
**Pull request by sssaiya**
## Summary
Add parking card using webview, as per Issue #771

## Changelog
[General] [Add] - Add parking card

## Test Plan
Verify card with parking data API at https://4pefyt8qv7.execute-api.us-west-2.amazonaws.com/dev/parking/v1.1/status/
Test on various low tier devices for latency issues.

### CODE REVIEW

File: lib/core/data_providers/cards_data_provider.dart [hunk 1]
```diff
@@ -24,6 +24,9 @@ class CardsDataProvider extends ChangeNotifier {
       'availability',
       'events',
       'news',
+//      'special_events',
```
Comment (i8wu) [inline lib/core/data_providers/cards_data_provider.dart:27]: commented out on purpose? If you've commented anything out as a part of dev/testing please revert them. If you're commenting something to save it for later. I'd recommend adding a todo so someone will remember to followup on it.

File: lib/core/data_providers/parking_data_provider.dart [hunk 2]
```diff
@@ -1,28 +1,38 @@
-import 'package:campus_mobile_experimental/core/data_providers/user_data_provider.dart';
+// import 'package:campus_mobile_experimental/core/data_providers/user_data_provider.dart';
```
Comment (i8wu) [inline lib/core/data_providers/parking_data_provider.dart:1]: comment

File: lib/core/data_providers/parking_data_provider.dart [hunk 3]
```diff
@@ -36,15 +46,24 @@ class ParkingDataProvider extends ChangeNotifier {
...
       ///replace old list of lots with new one
       _parkingModels = newMapOfLots;
 
+      //TODO Add user selected spots
+
       /// if the user is logged in we want to sync the order of parking lots amongst all devices
-      if (_userDataProvider != null) {
-        reorderLots(_userDataProvider.userProfileModel.selectedLots);
-      }
+      // if (_userDataProvider != null) {
```
Comment (i8wu) [inline lib/core/data_providers/parking_data_provider.dart:64]: comment

File: lib/core/data_providers/parking_data_provider.dart [hunk 4]
```diff
@@ -54,82 +73,73 @@ class ParkingDataProvider extends ChangeNotifier {
...
-  ///ADD A PARKING LOT IF IT HAS NOT ALREADY BEEN ADDED
-  void addLot(ParkingModel model) {
-    if (!_userDataProvider.userProfileModel.selectedLots
-        .contains(model.locationName)) {
-      _userDataProvider.userProfileModel.selectedLots.add(model.locationName);
+  ///RETURNS A List<ParkingModels> IN THE CORRECT ORDER
+  List<ParkingModel> get parkingModels {
+    ///check if we have an offline _parkingModel
+    if (_parkingModels != null) {
+      ///check if we have an offline _userProfileModel
```
Comment (i8wu) [inline lib/core/data_providers/parking_data_provider.dart:100]: comment

File: lib/core/data_providers/provider_setup.dart [hunk 5]
```diff
@@ -196,9 +196,10 @@ List<SingleChildWidget> dependentServices = [
       create: (_) {
     var parkingDataProvider = ParkingDataProvider();
     parkingDataProvider.fetchParkingLots();
+    parkingDataProvider.fetchSpotTypes();
     return parkingDataProvider;
   }, update: (_, userDataProvider, parkingDataProvider) {
-    parkingDataProvider.userDataProvider = userDataProvider;
+    //   parkingDataProvider.userDataProvider = userDataProvider;
```
Comment (i8wu) [inline lib/core/data_providers/provider_setup.dart:202]: comment

File: lib/core/navigation/router.dart [hunk 6]
```diff
@@ -136,6 +137,11 @@ class Router {
           return ScannerView();
         });
       case RoutePaths.ClassScheduleViewAll:
+        return MaterialPageRoute(builder: (_) => ClassList());
+      // case RoutePaths.ParkingLotsView:
```
Comment (i8wu) [inline lib/core/navigation/router.dart:141]: comment

File: lib/ui/views/parking/manage_parking_view.dart [hunk 7]
```diff
@@ -32,17 +32,24 @@ class ManageParkingView extends StatelessWidget {
     for (ParkingModel item in newOrder) {
       orderedLocationNames.add(item.locationName);
     }
-    parkingDataProvider.reorderLots(orderedLocationNames);
+    // parkingDataProvider.reorderLots(orderedLocationNames);
```
Comment (i8wu) [inline lib/ui/views/parking/manage_parking_view.dart:35]: comment

File: lib/core/data_providers/cards_data_provider.dart (hunk 1 above)
Comment (sssaiya) [reply lib/core/data_providers/cards_data_provider.dart]: Yup, thanks for catching those ! Have a bad practice of 'commenting out for later'. I have removed the extra comments in the most recent commit.

Comment (c3bryant) [reply lib/core/data_providers/cards_data_provider.dart]: The default cards API needed an update for the Parking card to activate. This has been completed in QA.

Re-testing on i8wu's branch, the Parking card is loading, but not the inner card content.

Err:
`Uncaught TypeError: Cannot read property 'split' of null", source: https://mobile.ucsd.edu/replatform/v1/qa/webview/parking/index.js (62)`

Comment (i8wu) [reply lib/core/data_providers/cards_data_provider.dart]: @c3bryant Looks like the "spots" are all deactivated by default. If you press manage spots and toggle them all on content shows up. I'm seeing an undefined text on it though 
image

Comment (sssaiya) [reply lib/core/data_providers/cards_data_provider.dart]: I have fixed the 2 problems discussed above. If lotContext is not provided then we can display a nbsp. And added a default message when no spot types are selected by user or provided in the url.

image

**Comment by sssaiya** [comment]
I have added the feature for persistent userPreferences selections, and have updated the branch with most recent changes from upstream as of now.

**Review by c3bryant — APPROVED** [review]

**Pull request merged**